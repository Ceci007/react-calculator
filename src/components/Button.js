import React from 'react';
import '../index.css';
import PropTypes from 'prop-types';

const Button = ({ name }) => (
  <button type="button">{ name }</button>
);

Button.propTypes = {
  name: PropTypes.string.isRequired,
};

export default Button;
