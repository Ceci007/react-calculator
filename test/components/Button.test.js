import React from 'react';
import renderer from 'react-test-renderer';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import Button from '../../src/components/Button';

let container = null;
const handleClick = jest.fn();
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('Button component', () => {
  test('should render with lightgrey default background color', () => {
    const button = renderer.create(
      <Button onClick={i => handleClick(i)} name="0" />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });

  test('should render with name equal to +', () => {
    const button = renderer.create(
      <Button onClick={i => handleClick(i)} name="+" />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });
});

it('renders with name property', () => {
  act(() => {
    render(<Button onClick={i => handleClick(i)} name="=" />, container);
  });

  expect(container.textContent).toBe('=');
});
