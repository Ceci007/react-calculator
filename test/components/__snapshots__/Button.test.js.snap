// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render with lightgrey default background color 1`] = `
<button
  className="btn"
  id="button-0"
  type="button"
>
  0
</button>
`;

exports[`Button component should render with name equal to + 1`] = `
<button
  className="btn"
  id="button-+"
  type="button"
>
  +
</button>
`;
